HLTrigger/HLTcore/interface/HLTPrescaleProvider.h

```diff
@@ -58,25 +59,53 @@ class HLTPrescaleProvider {
   // negative == error
 
   /// combining the two methods above
-  unsigned int prescaleValue(const edm::Event& iEvent, const edm::EventSetup& iSetup, const std::string& trigger);
-
+  template<typename T = unsigned int>
+  T prescaleValue(const edm::Event& iEvent, const edm::EventSetup& iSetup, const std::string& trigger){
+    const int set(prescaleSet(iEvent, iSetup));
+    return set < 0 ? -1 : 
```

Comment:
This changes the meaning as -1 was never returned, +1 was returned.

Reply:
ah good, thanks will fix to +1

Reply:
hmm this comes from prescaleValue and prescaleValues having different behaviour. 

prescaleValue() will return a 1 when there is an error while prescaleValues() will return -1. 

This I dont like and I think we should harmonise them to the same thing.  I'm sort of okay with `-1` converting to `unsigned int` given it is `std::numeric_limits<unsigned int>::max()` 

We might screw somebody over but we're depreciating ints and unsigned ints very soon anyways. If we do screw them over, at least their prescale will be very very large and obviously wrong.  Also this should be very very rare and basically should never happen to users and can only happen to folks re-running the HLT. Also it cant happen for just one trigger, it will be all or none. 

We could go the std::optional route as Andrea has suggested but I think that might be overkill. 

thoughts?

Reply:
-1 was used for the int return type used there, while the uint return type method used 1 - anyway I am ok with -1 for both....

Reply:
so I decided to specialise this method and have it return a 1 for an unsigned int. Given we're removing it soon, it doesnt matter

Reply:
I would rather propose to just drop `unsigned int` and use `int` and `-1` everywhere ?

Reply:
the plan is a second PR very soon which drops unsigned int and int support.  So lets leave this as a "changes nothing PR" and then we can have our breaking PR afterwards.